content/articles/creating-a-shopping-cart-api/index.md

```diff
@@ -0,0 +1,580 @@
+        }
+    }
+
+    @Override
+    public List<Product> checkAllTheProductsInTheCartByCustomer(Long id) {
+        List<Product> productList=new ArrayList<>();
+        if (customerRepo.getCustomer(id) != null) {
+            productList=cart.displayAllItems();
+        }
+            return productList;
+        }
+
+    @Override
+    public List<Product> checkAllTheProductsInTheCartByAdmin(Long id) {
+        List<Product> productList=new ArrayList<>();
+        if (adminDao.getAdmin(id) != null) {
+            productList=cart.displayAllItems();
+        }
+        return productList;
+    }
+
+
+    @Override
+    public BigDecimal calculatePrice() {
+        return cart.calculatePriceInCart();
+    }
+
+
+    @Override
+    public int totalNumberOfItems() {
+        return cart.totalNumberOfProducts();
+    }
+
+    @Override
+    public BigDecimal calculatingPurchase(Long id) {
+        return customerRepo.getCustomer(id).getBalance().subtract(calculatePrice());
+    }
+}
+```
+#### DataServiceImpl
+```java
+@Service
+public class DataServiceImpl implements dataService{
+    @Autowired
+    private CustomerDao customerRepo;
+    @Autowired
+    private AdminDao adminRepo;
+    @Override
+    public Customer registerCustomer(Customer customer) {
+
+        if(customerRepo.getCustomer(customer.getId())==null){
+           customer= customerRepo.registerCustomer(customer.getId(),customer);
+        }
+        else
+            System.out.println("Customer Already Exist");
+        return customer;
+    }
+
+
+    @Override
+    public Admin registerAdmin(Admin admin) {
+
+    if(adminRepo.getAdmin(admin.getId())==null) {
+
+        admin = adminRepo.registerAdmin(admin.getId(), admin);
+    }
+       else
+        System.out.println("Admin Already Exist");
+        return admin;
+
+    }
+
+    @Override
+    public List<Customer> gettingAllCustomers() {
+        List<Customer> customerList=new ArrayList<>();
+        for(int counter=1;counter<=customerRepo.gettingAllTheCustomers().size();counter++){
+            Long l2= (long) counter;
+            customerList.add(customerRepo.getCustomer(l2));
+        }
+        return customerList;
+    }
+
+    @Override
+    public void removeCustomer(Long id) {
+        if(customerRepo.getCustomer(id)==null){
+            System.out.println("Customer does not exist");
+        }
+        else
+            customerRepo.removeCustomer(id);
+    }
+
+    @Override
+    public void removeAdmin(Long id) {
+        if(adminRepo.getAdmin(id)==null){
+            System.out.println("Customer does not exist");
+        }
+        else
+            adminRepo.removeAdmin(id);
+    }
+
+    @Override
+    public List<Admin> getAllAdmin() {
+        List<Admin> adminList=new ArrayList<>();
+        for(int counter=1;counter<=adminRepo.getAllAdmin().size();counter++){
+            Long l2= (long) counter;
+            adminList.add(adminRepo.getAdmin(l2));
+        }
+        return adminList;
+    }
+}
```

Comment:
Please reformat this code, notably, the lack of curly braces surrounding the **else** clauses.